### PR TITLE
Make "Hide reviewed" considerate: animate, undo, and surface recents

### PR DESCRIPTION
**Pull request by fgilio**
Three layered improvements so marking a file reviewed no longer feels
abrupt or hard to recover from:

- Mark-reviewed undo toast (⌘Z), routed through the existing
  undo-available stack with a 3s coalesce window so power users marking
  many files in a row see one merged toast.
- Fade + height-collapse on file wrappers (sidebar row + main content)
  with a global prefers-reduced-motion fallback in app.css.
- "Recently reviewed" sidebar group: last 5 LIFO files stay visible,
  dimmed and struck, with an in-place un-mark control. Only shown in
  Hide-reviewed mode; cleared on Show-all and via an explicit Clear.

Resolves the diff-file reset-reviewed-files debt by adding the listener
on diff-file's root.

Documents the underlying UX principles (motion durations/easing,
reversibility, coalescing, accessibility, Doherty threshold) in
resources/CLAUDE.md so we have a checklist for future audits across
the rest of the app.

## Summary by CodeRabbit

* **New Features**
  * "Recently reviewed" sticky sidebar showing a capped, deduped MRU of recently marked files
  * New undo event to revert reviewed files; undo now removes reviewed records and updates UI

* **Accessibility**
  * Respect for users' reduced-motion preference (animations/transitions effectively disabled)

* **Bug Fixes**
  * Improved sync between reviewed state, undo, and UI mirrors

* **Tests**
  * Comprehensive tests covering undo, MRU behavior, and reviewed-state edge cases

### CODE REVIEW

**Comment by fgilio** [comment]
@codex review

File: resources/views/pages/⚡review-page.blade.php [hunk 1]
```diff
@@ -704,17 +716,71 @@ public function undo(string $type, mixed $payload): void
...
+            $reverted,
+        )));
+
+        $this->recentlyReviewedIds = array_values(array_filter(
+            $this->recentlyReviewedIds,
+            fn (string $id): bool => ! in_array($id, $revertedIds, true),
+        ));
+
+        $this->dispatch('reviewed-files-reverted', fileIds: $revertedIds);
+    }
```
[the remark this answers is not in the document]
Comment (fgilio) [reply resources/views/pages/⚡review-page.blade.php:759]: Fixed in 5a651f3 — `unmarkReviewed()` now calls `skipRender()` after the dispatch. Same rationale as `toggleReviewed`: the sidebar `recentlyReviewedIds` is read reactively via `$wire`, the file-row check mirrors `reviewedFiles[id]` via `:class`, and the `reviewed-files-reverted` event flips DiffFile's local `reviewed` mirror — no Blade output depends on the new server state, so a re-render would just re-hydrate every child for nothing.

Test added: `unmarkReviewed skips parent re-render to avoid 1+N child hydration`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: resources/views/livewire/undo-toast.blade.php [hunk 2]
```diff
@@ -12,11 +13,30 @@
...
+                && top && top.type === 'mark-reviewed'
+                && (Date.now() - top.createdAt) < this.coalesceWindowMs
+                && Array.isArray(detail.payload?.filePaths)
+                && Array.isArray(top.payload?.filePaths)
+            ) {
+                top.payload.filePaths.push(...detail.payload.filePaths);
+                top.createdAt = Date.now();
+                top.expiresAt = Date.now() + ttl * 1000;
+                const n = top.payload.filePaths.length;
+                top.message = n + ' files marked as reviewed';
```
[the remark this answers is not in the document]
Comment (fgilio) [reply resources/views/livewire/undo-toast.blade.php]: Fixed in 5a651f3 — adopted the suggested `Array.from(new Set(...))` dedupe and pluralized the message ("1 file" / "N files"). Sequence mark `Foo` → un-mark `Foo` → mark `Foo` now coalesces to a single entry with `filePaths: ['Foo.php']` and a "1 file marked as reviewed" message.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: resources/views/pages/⚡review-page.blade.php [hunk 3]
```diff
@@ -729,6 +795,42 @@ public function toggleReviewed(string $filePath): void
         }
 
         $this->reviewedFiles = $result;
+
+        $isNowReviewed = array_key_exists($filePath, $this->reviewedFiles);
+        $fileId = collect($this->files)->firstWhere('path', $filePath)['id'] ?? null;
+
+        if (! $wasReviewed && $isNowReviewed) {
+            if ($fileId !== null) {
+                $this->recentlyReviewedIds = array_slice(
```
[the remark this answers is not in the document]
Comment (fgilio) [reply resources/views/pages/⚡review-page.blade.php]: Good catch — fixed in 5a651f3. The `$fileId` lookup in `toggleReviewed()` now scans `$this->sourceFiles` instead of `$this->files`, so review-pair `.json/.md` artifacts never enter `recentlyReviewedIds` (and therefore can't push real source files out of the 5-slot MRU). Test added: `marking a review-pair artifact does not consume a "Recently reviewed" slot`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: resources/views/pages/⚡review-page.blade.php [hunk 4]
```diff
@@ -704,17 +716,71 @@ public function undo(string $type, mixed $payload): void
...
+            return;
+        }
+
+        $action = app(ToggleReviewedAction::class);
+        $target = $this->buildDiffTarget();
+        $reverted = [];
+
+        foreach ($filePaths as $filePath) {
+            if (! array_key_exists($filePath, $this->reviewedFiles)) {
+                continue;
```
[the remark this answers is not in the document]
Comment (fgilio) [reply resources/views/pages/⚡review-page.blade.php]: Fixed in 5a651f3. `unmarkReviewed()` now does a direct `whereIn('file_path', $filePaths)->delete()` against `ReviewedFile`, scoped to the project/repo via the existing `forProjectOrRepo` scope. The toast's `filePaths` is treated as the source of truth — refreshes / hash-drift can no longer cause undo to no-op while leaving the DB row behind.

Test added: `undo mark-reviewed deletes the DB row even when reviewedFiles map dropped the entry (refresh / hash drift)` — sets `reviewedFiles` to `[]` after marking, then calls undo and asserts the DB row is gone.

---
_Generated by [Claude Code](https://claude.ai/code)_
